lightly/api/api_workflow_datasets.py

```diff
@@ -166,18 +252,21 @@ def create_new_dataset_with_unique_name(
                 constants `DatasetType.IMAGES` and `DatasetType.VIDEOS`.
 
         """
-        current_datasets = self.get_datasets()
-        current_datasets_names = [dataset.name for dataset in current_datasets]
-
-        if dataset_basename not in current_datasets_names:
+        if not self.dataset_name_exists(dataset_name=dataset_basename):
             self._create_dataset_without_check_existing(
                 dataset_name=dataset_basename,
                 dataset_type=dataset_type
             )
         else:
+            existing_datasets = self._datasets_api.get_datasets_query_by_name(
+                dataset_name=dataset_basename,
+                exact=False,
+                shared=False,
+            )
+            existing_dataset_names = {dataset.name for dataset in existing_datasets}
             counter = 1
             dataset_name = f"{dataset_basename}_{counter}"
-            while dataset_name in current_datasets_names:
+            while dataset_name in existing_dataset_names:
```

Comment:
a bit hacky as the returned datasetNames don't need to be numerical but e.g. `hello_1` und `hello_guarin`. But I get why you are doing it this way - better safe then sorry :)

Reply:
Yes, but we don't have a specific endpoint for datasets with numerical suffixes 😇

Reply:
in theory you could regex the pattern and count ;) but all fine in my eyes :)